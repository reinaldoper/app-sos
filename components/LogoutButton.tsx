import React from 'react';
import { StyleSheet, Text, Alert, TouchableOpacity } from 'react-native';
import { useRouter } from 'expo-router';

export default function LogoutButton() {
  const router = useRouter();

  const handleLogout = async () => {
    try {
      router.replace('/Logout'); 
    } catch (error: any) {
      Alert.alert("Erro ao sair", error.message);
    }
  };

  return (
    <TouchableOpacity style={styles.button} onPress={handleLogout}>
      <Text style={styles.text}>ðŸšª Logout</Text>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  button: {
    backgroundColor: '#e53935',
    padding: 12,
    alignItems: 'center',
  },
  text: {
    color: '#fff',
    fontWeight: 'bold',
    fontSize: 16,
  },
});
