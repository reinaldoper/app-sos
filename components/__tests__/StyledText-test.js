import * as React from 'react';
import renderer from 'react-test-renderer';
import LogoutButton from '../LogoutButton';

it(`renders correctly`, () => {
  const tree = renderer.create(<LogoutButton>Logout</LogoutButton>).toJSON();

  expect(tree).toMatchSnapshot();
  expect(tree).toHaveProperty('type', 'View');
  expect(tree).toHaveProperty('props.onClick');
  expect(tree.props.onClick).toBeInstanceOf(Function);
});

