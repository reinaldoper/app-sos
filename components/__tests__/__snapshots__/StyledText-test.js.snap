// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#e53935",
      "opacity": 1,
      "padding": 12,
    }
  }
>
  <Text
    style={
      {
        "color": "#fff",
        "fontSize": 16,
        "fontWeight": "bold",
      }
    }
  >
    ðŸšª Logout
  </Text>
</View>
`;
